test/cppsuite/tests/search_near_02.cxx

```diff
@@ -0,0 +1,430 @@
+     */
+    void
+    search_near(
+      scoped_cursor &cursor_default, scoped_cursor &cursor_prefix, const std::string &prefix)
+    {
+        /* Call search near with both cursors using the given prefix. */
+        cursor_default->set_key(cursor_default.get(), prefix.c_str());
+        cursor_prefix->set_key(cursor_prefix.get(), prefix.c_str());
+
+        int exact_default, exact_prefix;
+        int ret_default = cursor_default->search_near(cursor_default.get(), &exact_default);
+        int ret_prefix = cursor_prefix->search_near(cursor_prefix.get(), &exact_prefix);
+
+        /*
+         * It is not possible to have a prefix search near call successful and the default search
+         * near call unsuccessful.
+         */
+        testutil_assert(
+          ret_default == ret_prefix || (ret_default == 0 && ret_prefix == WT_NOTFOUND));
+
+        /* We only have to perform validation when the default search near call is successful. */
+        if (ret_default == 0) {
+            /* Both calls are successful. */
+            if (ret_prefix == 0)
+                validate_successful_calls(
+                  cursor_default, cursor_prefix, prefix, exact_default, exact_prefix);
+            /* The prefix search near call failed. */
+            else
+                validate_unsuccessful_prefix_call(cursor_default, prefix, exact_default);
+        }
+    }
+
+    void
+    validate_successful_calls(scoped_cursor &cursor_default, scoped_cursor &cursor_prefix,
+      const std::string &prefix, int exact_default, int exact_prefix)
+    {
+        const char *k;
+        std::string k_str;
+        int ret;
+
+        /*
+         * The prefix search near call cannot retrieve a key with a smaller value than the prefix we
+         * searched.
+         */
+        testutil_assert(exact_prefix == 0 || exact_prefix == 1);
+
+        /* Retrieve the keys each cursor is pointing at. */
+        const char *key_default;
+        testutil_check(cursor_default->get_key(cursor_default.get(), &key_default));
+        std::string key_default_str = key_default;
+
+        const char *key_prefix;
+        testutil_check(cursor_prefix->get_key(cursor_prefix.get(), &key_prefix));
+        std::string key_prefix_str = key_prefix;
+
+        /* The key at the prefix cursor should contain the prefix. */
+        testutil_assert(key_prefix_str.substr(0, prefix.size()) == prefix);
+
+        logger::log_msg(LOG_TRACE,
+          "search_near (normal) exact " + std::to_string(exact_default) + " key " + key_default);
+        logger::log_msg(LOG_TRACE,
+          "search_near (prefix) exact " + std::to_string(exact_prefix) + " key " + key_prefix);
+
+        /* Example: */
+        /* keys: a, bb, bba */
+        /* Default search_near(bb) returns a, exact = -1 */
+        /* Prefix search_near(bb) returns bb, exact = 0 */
+        /* Prefix search_near(bb) returns bba, exact = 1 */
+        if (exact_default == -1) {
```

Comment:
We don't need to worry if prefix search near returns back "bba", since "bb" should be set always.

Reply:
What if `bb` is not visible?

Reply:
I am thinking here that we only need to check for the next one, we don't need a while loop, do we ?
If `bb` is not visible, `bba` will be the next one and the prefix cursor should be pointing at it, right ?

Reply:
yeah I agree `bba` should be the next one. Don't think we need a while loop in this case.

Reply:
Fixed in a5602ced362d3cf2d7bf95b2e4b8469689f4265a